crates/hotshot/src/traits/networking/web_server_network.rs

```diff
@@ -768,8 +770,6 @@ impl<TYPES: NodeType + 'static> ConnectedNetwork<Message<TYPES>, TYPES::Signatur
         );
 
         // TODO ED Need to handle canceling tasks that don't receive their expected output (such a proposal that never comes)
-        // TODO ED Need to GC all old views, not just singular views, could lead to a network leak
-
         match event {
             ConsensusIntentEvent::PollForProposal(view_number) => {
```

Comment:
I think we should also GC the view sync stuff here, it's possible we just don't hit view sync for a long time, and we'd be running the view sync polling task for a ton of views.  In fact if maybe we can just cancel all old task regardless of what we are getting the new polling intent for.  Basically every time we start to poll for a new view number we can GC old view task from all the task maps

Reply:
We can do this but I have to be careful; certain things like transactions we poll way ahead for. If we start polling for txes in view n+3, polls for other things in view n would get cancelled before we have everything resolved

Reply:
Agree that's a good point we need to be careful.  I was more thinking only for the big things, like proposal and view sync certificates.  On second thought maybe it's best to have a specific event like `CancelOldPolls` which we inject when we update our view that cancels the old view polling.  That way we have one entry point for GCing but don't have to piggy back on the old events which might have weirdness like you mentioned.